Src/BootCamp.Chapter/ToggleableGrid2d.cs

```diff
@@ -1,16 +1,52 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+
+namespace BootCamp.Chapter
 {
     public class ToggleableGrid2D : IToggleableGrid
     {
         private readonly IGridClearer _gridClearer;
+        private readonly bool[,] _toggles;
 
         public ToggleableGrid2D(bool[,] toggles, IGridClearer gridClearer)
         {
             _gridClearer = gridClearer;
+            _toggles = toggles;
         }
 
         public void Toggle(int x, int y)
         {
+            var current = _toggles[x, y];
+            _toggles[x, y] = !current;
+            Redraw(_toggles);
+        }
+
+        private void Redraw(bool[,] toggles)
+        {
+            Console.OutputEncoding = System.Text.Encoding.Unicode;
+            var numberOfRows = toggles.GetLength(0);
+            var numberOfCols = numberOfRows;
```

Comment:
In the assignment, it was said that rows and columns can have different sizes. In your case, you presume it's a square, which is not the case.

Reply:
I thought in a 2DGrid the colums and rows schould be the same.

Reply:
That's not what it means. 2D array has all rows with the same length, but that doesn't mean that columns length and rows length will be matching.

Reply:
will change that